extension/agenthealth/handler/stats/provider/statuscode.go

```diff
@@ -0,0 +1,216 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package provider
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/amazon-contributing/opentelemetry-collector-contrib/extension/awsmiddleware"
+
+	"github.com/aws/amazon-cloudwatch-agent/extension/agenthealth/handler/stats/agent"
+)
+
+const (
+	statusResetInterval = 5 * time.Minute
+	statusHandlerID     = "cloudwatchagent.StatusCodeHandler"
+	channelBufferSize   = 100
+)
+
+var (
+	statusCodeProviderSingleton *StatsProvider
+	statsProviderOnce           sync.Once
+)
+
+// StatsProvider handles processing of status codes and maintains stats.
+type StatsProvider struct {
```

Comment:
nit: Rename to something more specific
```suggestion
type StatusCodeProvider struct {
```